### PR TITLE
SRCH-3803 AffiliateIndexedDocumentFetcher rescue

### PR DESCRIPTION
**Pull request by lorawoodford**
## Summary
- Response to the comment here: https://github.com/GSA/search-gov/pull/1124/files#r1102881459
- Adds an additional `rescue` to confirm all documents in a batch that can be fetched, will be fetched, even if other documents have disappeared between when the job was queued and when it was executed.
- I opted to add an additional `rescue` rather than move the existing one because - based on interrogating the test currently in main - I think we're trying to (and should) catch two different scenarios here.  First, an instance where the `Affiliate` is gone (that's currently caught in main), but second, an instance where the IndexedDocument itself is gone (that's what this PR addresses).
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

**Review by MyNameIsMissing — DISMISSED** [review]
LGTM

Admittedly I'm not qualified to review the spec, but I get what the rescue is doing.  Since you'll have to merge main in again before you can merge this, maybe have someone else double check the spec.

File: spec/jobs/affiliate_indexed_document_fetcher_job_spec.rb [hunk 1]
```diff
@@ -61,9 +61,42 @@
...
+    let(:unfetched2) { IndexedDocument.find_by(unfetched_atts2) }
+    let(:unfetched_atts2) do
+      { url: 'http://nps.gov/foo2.html',
+        title: 'Doc Title 2',
+        description: 'This is a document 2.' }
+    end
+
+    before do
+      affiliate.indexed_documents.build(unfetched_atts2)
+      affiliate.save
```
Comment (MothOnMars) [inline spec/jobs/affiliate_indexed_document_fetcher_job_spec.rb:74]: I recommend simplifying these two lines as:
```ruby
affiliate.indexed_documents.create!(unfetched_atts2)
```
I prefer to use the bang methods in specs to ensure that an error will be raised if the spec setup is faulty. However, this is just a suggestion, so I'll go ahead and approve this PR.

**Review by MothOnMars — DISMISSED** [review]
LGTM. Thanks for the comment re. the original `rescue`'s purpose.

File: spec/jobs/affiliate_indexed_document_fetcher_job_spec.rb (hunk 1 above)
Comment (lorawoodford) [reply spec/jobs/affiliate_indexed_document_fetcher_job_spec.rb:74]: Good call.  Change made.

**Review by MothOnMars — APPROVED** [review]
LGTM

**Pull request merged**